examples/chatmemberbot.py

```diff
@@ -0,0 +1,167 @@
+#!/usr/bin/env python
+# pylint: disable=C0116
+# This program is dedicated to the public domain under the CC0 license.
+
+"""
+Simple Bot to handle '(my_)chat_member' updates.
+Greets new users & keeps track of which chats the bot is in.
+
+Usage:
+Press Ctrl-C on the command line or send a signal to the process to stop the
+bot.
+"""
+
+import logging
+from typing import Tuple, Optional
+
+from telegram import Update, Chat, ChatMember, ParseMode, ChatMemberUpdated
+from telegram.ext import (
+    Updater,
+    CommandHandler,
+    CallbackContext,
+    ChatMemberHandler,
+)
+
+# Enable logging
+logging.basicConfig(
+    format="%(asctime)s - %(name)s - %(levelname)s - %(message)s", level=logging.INFO
+)
+
+logger = logging.getLogger(__name__)
+
+
+def extract_status_change(
+    chat_member_update: ChatMemberUpdated,
+) -> Optional[Tuple[bool, bool]]:
+    """Takes a ChatMemberUpdated instance and extracts whether the 'old_chat_member' was a member
+    of the chat and whether the 'new_chat_member' is a member of the chat. Returns None, if
+    the status didn't change."""
+    status_change = chat_member_update.difference().get("status")
+    old_is_member, new_is_member = chat_member_update.difference().get("is_member", (None, None))
+
+    if status_change is None:
+        return None
+    # We first check if the bot was a chat member before and if it it now
+    old_status, new_status = status_change
+    was_member = (
+        old_status
+        in [
+            ChatMember.MEMBER,
+            ChatMember.CREATOR,
+            ChatMember.ADMINISTRATOR,
+        ]
+        or (old_status == ChatMember.RESTRICTED and old_is_member is True)
```

Comment:
mh, not quite. We'd still have to check for the restricted part (at least that's how I understand the docs), so that would only change the first condition slightly

Reply:
that `or` condition checks if the member is restricted and is still in the group, right? Cause if they left, then it should be updated to `ChatMember.LEFT` regardless of their restriction.

Reply:
But if `ChatMember.RESTRICTED` + `is_member=False` is no option, then why is there a `is_member` attribute in the first place?

Reply:
uh, idk. Gotta test that and ask the bot api devs maybe

Reply:
Already tried :D kick a member and then restrict them. In the App that lists them not in the "removed users" list but in the "exceptions" list

Reply:
ah. That's just weird behaviour. Guess it's fine to leave that as it is then.